modules/web3/helpers/errors.ts

```diff
@@ -1,9 +1,27 @@
-export const TX_NOT_ENOUGH_FUNDS = "sender doesn't have enough funds to send tx";
+import { getMessageFromCode } from 'eth-rpc-errors';
 
-export function parseTxError(_error): string {
-  const error = _error.message ? _error.message : _error;
-  if (error.includes("sender doesn't have enough funds to send tx")) {
+export const TX_NOT_ENOUGH_FUNDS = "Sender doesn't have enough funds to send the transaction";
+export const USER_REJECTED = 'User rejected the transaction';
+
+export function parseTxError(error: Error): string {
+  const defaultError = 'Something went wrong with your transaction.';
+
+  const message = error.message ? error.message : defaultError;
+  if (message.includes("sender doesn't have enough funds to send tx")) {
     return TX_NOT_ENOUGH_FUNDS;
   }
-  return error;
+
+  if (message.includes('user rejected transaction')) {
```

Comment:
also why not utilize the error code here? In case the message changes someday. We can still return a custom message if that's the goal. `if (error.code === 4001) myCustomMessage`